src/axom/sidre/tests/sidre_collections.cpp

```diff
@@ -0,0 +1,582 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/// Tests sidre's ItemCollection hierachy,
+/// including MapCollection, ListCollection and IndexedCollection
+
```

Comment:
Note: sidre's `ItemCollection` hierarchy was previously untested.
This file covers tests for all derived classes of `ItemCollection`